### DISCUSSION TITLE
Loading different versions of tailwind.config depending on environment

### DISCUSSION THREAD

**Discussion by colinmj** [tailwindlabs/tailwindcss#7494] [Help]
Is it possible to load a certain version of the tailwind.config based on dynamic data, such as an env variable? I have a Vue js app built with tailwind that I would like to look unique (different colors and fonts) to different sets of users. I would like to avoid having to write conditional classes for the entire application, ideally I would have 2 config files with all the same variables but different values, and load one dynamically depending on the environment variable

**Comment by dotnetCarpenter** [comment]
It's a js file so you can `module.exports` whatever you want. I think this is more of question about build environment. What build tool do you use with vuejs?

If you use webpack, you can use `process.env`: https://webpack.js.org/api/cli/#configuration-options

```js
module.export = process.env.tailwind == 1 ? require('./tailwind.config1.js') : require('./tailwind.config2.js')
```
_tailwind.config.js_

```
npx webpack --env tailwind=2
```
_Command line to load tailwind.config2.js_

Change names and variables as you wish :)

**Comment by timwuhaotian** [comment]
I guess you can also use cross-env to bake some variable to indicate which build mode you are running, for example with vite + tailwind:

`cross-env BUILD_MODE=amp vite build` and read it in tailwind config as `process.env.BUILD_MODE` return different configs accordingly
